### ISSUE TITLE
delete matomo cookies

### ISSUE THREAD

**Issue by giannisdag** [KIProtect/klaro#56]
Hi,
I am trying to delete matomo cookies but I cannot achieve that following the guide. I can only add the matomo cookies using the guide. I also tried the example with the inline tracker, and have the same results. I am also adding the config file. Also in https://klaro.kiprotect.com/ where you are using matomo, **it also does not seem to work**.
`
// By default, Klaro will load the config from  a global "klaroConfig" variable.
// You can change this by specifying the "data-config" attribute on your
// script take, e.g. like this:
// <script src="klaro.js" data-config="myConfigVariableName" />
// You can also disable auto-loading of the consent notice by adding
// data-no-auto-load=true to the script tag.
var klaroConfig = {

    // You can customize the ID of the DIV element that Klaro will create
    // when starting up. If undefined, Klaro will use 'klaro'.
    elementID: 'klaro',

    // You can customize the name of the cookie that Klaro uses for storing
    // user consent decisions. If undefined, Klaro will use 'klaro'.
    cookieName: 'klaro',

    // You can also set a custom expiration time for the Klaro cookie.
    // By default, it will expire after 120 days.
    cookieExpiresAfterDays: 365,

    // You can customize the name of the cookie that Klaro will use to
    // store user consent. If undefined, Klaro will use 'klaro'.
 
    // Put a link to your privacy policy here (relative or absolute).
    privacyPolicy: '/site/cookie-policy',
    
    // Defines the default state for applications (true=enabled by default).
    default: true,

    // If "mustConsent" is set to true, Klaro will directly display the consent
    // manager modal and not allow the user to close it before having actively
    // consented or declines the use of third-party apps.
    mustConsent: false,

    // You can define the UI language directly here. If undefined, Klaro will
    // use the value given in the global "lang" variable. If that does
    // not exist, it will use the value given in the "lang" attribute of your
    // HTML tag. If that also doesn't exist, it will use 'en'.
    //lang: 'en',

    // You can overwrite existing translations and add translations for your
    // app descriptions and purposes. See `src/translations.yml` for a full
    // list of translations that can be overwritten:
    // https://github.com/DPKit/klaro/blob/master/src/translations.yml

    // Example config that shows how to overwrite translations:
    // https://github.com/DPKit/klaro/blob/master/src/configs/i18n.js
    translations: {
        // If you erase the "consentModal" translations, Klaro will use the
        // defaults as defined in translations.yml
        el: {
            consentModal: {
                title: 'Διαχείρισης πολιτικής απορρήτου',
                description: 'Εδώ μπορείτε να κάνετε επιλογές σχετικά με την πολιτική απορρήτου',
                privacyPolicy : {
                    name : 'Πολιτική απορρήτου',
                    text : 'Για περισσότερες πληροφορίες, παρακαλώ διαβάστε την {privacyPolicy}.'
                } 
            },
            consentNotice:{
                changeDescription : 'Πραγματοποιήθηκαν αλλαγές μετά την τελευταία σας επίσκεψη παρακαλούμε ανανεώστε την συγκατάθεση σας',
                description : 'Συγκεντρώνουμε και επεξεργαζόμαστε τα προσωπικά δεδομένα σας για τους παρακάτω σκοπούς: {purposes}.',
                learnMore : 'Περισσότερα'
            },
            decline: 'Απόρριπτω',
            ok: 'Συμφωνώ',
            save: 'Αποθήκευση',
            close: 'Κλείσιμο',
            app: {
                disableAll: {
                    title: 'Για όλες τις εφαρμογές',
                    description: 'Χρησιμοποίησε αυτό τον διακόπτη για να ενεργοποιήσεις/απενεργοποιήσεις όλες τις εφαρμογές',
                },
                optOut: {
                    title: '(Μη απαιτούμενο)',
                    description: 'Είναι προκαθορισμένο να φορτώνεται, άλλα μπορεί να παραληφθεί',
                },
                required: {
                    title: '(Απαιτούμενο)',
                    description: 'Δεν γίνεται να λειτουργήσει σωστά η εφαρμογή χωρίς αυτό'
                },
                purposes: 'Σκοποί',
                purpose: 'Σκοπός'
            },
            poweredBy: 'Υποστηρίζεται από το Klaro',
            matomo: {
                description: 'Σύστημα παρακολούθησης επισκεψιμότητας',
            },
            akadim_security: {
                description: 'cookie ασφάλειας'
            },
            akadim_session: {
                description: 'Παρακολούθηση λειτουργίας'
            },
            cloudflare: {
                description: 'Προστασία από κυβερνοεπιθέσεις (DDoS)',
            },
            purposes: {
                analytics: 'Στατιστικά επισκεψιμότητας',
                security: 'Ασφάλεια',
                functionality: 'Λειτουργικότητα'
            },
        },
        en: {
            consentModal: {
                description: 'Here you can see and customize the information that we collect about you. Entries marked as "Example" are just for demonstration purposes and are not really used on this website.',
            },
            akadim_session: {
                description: 'Example of an inline tracking script',
            },
            matomo: {
                description: 'Collecting of visitor statistics',
            },
            cloudflare: {
                description: 'Protection against DDoS attacks',
            },
            purposes: {
                analytics: 'Analytics',
                security: 'Security',
                livechat: 'Livechat',
                advertising: 'Advertising',
                styling: 'Styling',
            }
        },
    },

    // This is a list of third-party apps that Klaro will manage for you.
    apps : [
        {
            // Each app should have a unique (and short) name.
            name : 'matomo',

            // If "default" is set to true, the app will be enabled by default
            // Overwrites global "default" setting.
            // We recommend leaving this to "false" for apps that collect
            // personal information.
            // default: true,

            // The title of you app as listed in the consent modal.
            title : 'Matomo',

            // The purpose(s) of this app. Will be listed on the consent notice.
            // Do not forget to add translations for all purposes you list here.
            purposes : ['analytics'],
            
            // A list of regex expressions or strings giving the names of
            // cookies set by this app. If the user withdraws consent for a
            // given app, Klaro will then automatically delete all matching
            // cookies.
            cookies : [
                // you can also explicitly provide a path and a domain for
                // a given cookie. This is necessary if you have apps that
                // set cookies for a path that is not "/" or a domain that
                // is not the current domain. If you do not set these values
                // properly, the cookie can't be deleted by Klaro
                // (there is no way to access the path or domain of a cookie in JS)
                '_pk_'
                // /^_pk_.*$/i //for the production version
                // [/^_pk_*$/, '/', 'localhost'], //for the local version
            ],
            
            // An optional callback function that will be called each time
            // the consent state for the app changes (true=consented). Passes
            // the `app` config as the second parameter as well.
            callback : function(consent, app){
                // This is an example callback function.
                // if (consent==false) {
                //     var _paq = _paq || [];
                //     _paq.push(['disableCookies']);    
                // }
            },

            // If "required" is set to true, Klaro will not allow this app to
            // be disabled by the user.
            // required : false,

            // If "optOut" is set to true, Klaro will load this app even before
            // the user gave explicit consent.
            // We recommend always leaving this "false".
            // optOut : false,

            // If "onlyOnce" is set to true, the app will only be executed
            // once regardless how often the user toggles it on and off.
            // onlyOnce: false,
        },

        // The apps will appear in the modal in the same order as defined here.
        {
            name : 'akadim_security',
            title : 'Akadim',
            purposes : ['security'],
            cookies : [
                ['cookie_new']
            ],
            // required: true,
        },
        {
            name : 'akadim_session',
            default: true,
            title : 'Akadim',
            purposes : ['security','functionality'],
            cookies : [
                'PHPFRONTSESSID'
            ],
            required: true,
        },
        {
            name : 'cloudflare',
            title : 'Cloudflare',
            purposes : ['security'],
            required: true,
        },
    ],
}`

**Comment by adewes** [comment]
Hey Giannis, I can confirm that sometimes Cookies are not properly deleted (e.g. when the domain or path does not match). I built a new version of Klaro with a more "agressive" cookie deletion routine that will try to delete the cookie without path and domain first. Seems that some browsers (notably Chrome) don't match cookies that were created without an explicit domain when you explicitly specify the domain, even when the two match.
Let me know if it works better now (might take 10-15 minutes for the new version to be up)!

**Comment by giannisdag** [comment]
Yes, it is working now thanks. I only added the path not the domain, for example [/^_pk_.*$/,'/'], as in my case it is a multisite, so the domain has to be irrelevant. Also I noticed that in chrome it deletes the cookies after refreshing the page, but in firefox the deletion happens instantly. I did not tested yet in IE, Edge and other browser types

**Comment by amenk** [comment]
@adewes We have a similar problem (not with Klaro), can you give me a hint where this "agressive cookie deletion code is located?
